go/vt/vtctl/workflow/common/utils.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2024 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"vitess.io/vitess/go/vt/log"
+	"vitess.io/vitess/go/vt/topo"
+)
+
+// GetShards returns a subset of shards in a keyspace. If no subset is provided, all shards are returned.
```

Comment:
It really only validates the subset, right? We're returning the same underlying string array values, aren't we?

Reply:
Not really, it is a helper function that also wraps the call to the topo to get all the shards in the keyspace. It can be called with a nil or non-nil subset.